src/constructApplications.js

```diff
@@ -119,17 +119,32 @@ function placeLoader(appName, appRoute, loadingPromise) {
     // We need the application container element to place the loader into
     const htmlId = applicationElementId(appName);
     let applicationElement = document.getElementById(htmlId);
-    let hidElement;
+    let makeElementVisible;
 
     if (!applicationElement) {
       applicationElement = document.createElement("div");
       applicationElement.id = htmlId;
       // Wait for layout engine to place this dom element in correct location
       // before it's visible
       applicationElement.style.display = "none";
-      hidElement = true;
 
       document.body.appendChild(applicationElement);
+
+      makeElementVisible = () => {
```

Comment:
Previously we were hiding the application element (line 129) and were only unhiding once it was the app code was downloaded. That defeats the purpose of the loader. The reason we are hiding it is we need to make sure it is positioned correctly in the dom before showing it, which only occurs during the single-spa:before-mount-routing-event. So what I've done is unhide it during that event.